src/Provider/GoogleMaps/GoogleMaps.php

```diff
@@ -267,6 +269,19 @@ private function updateAddressComponent(AddressBuilder $builder, string $type, $
                 $builder->addAdminLevel(intval(substr($type, -1)), $values->long_name, $values->short_name);
                 break;
 
+            case 'sublocality_level_1':
+            case 'sublocality_level_2':
+            case 'sublocality_level_3':
+            case 'sublocality_level_4':
+            case 'sublocality_level_5':
+                $newSubLocalityLevel = [
+                    'level' => intval(substr($type, -1)),
+                    'name' => $values->long_name,
+                    'code' => $values->short_name,
+                ];
+                $builder->setValueInArray('subLocalityLevel', $newSubLocalityLevel);
```

Comment:
Something like this: 

```php
function updateAddressComponent() {
   $newSubLocalityLevel = [];
   switch ($type) {
      // ...
      case 'sublocality_level_1':
      case 'sublocality_level_2':
      case 'sublocality_level_3':
      case 'sublocality_level_4':
      case 'sublocality_level_5':
           $newSubLocalityLevel[] = [
                  'level' => intval(substr($type, -1)),
                  'name' => $values->long_name,
                  'code' => $values->short_name,
            ];
    } // End of switch
    $builder->setValue('subLocalityLevel', $newSubLocalityLevel);
}
```

With the code above you do not need the `setValueInArray` function.